### PR TITLE
Optimize plugin's models

### PR DESCRIPTION
**Pull request by shargon**
# Description

Partial copy of https://github.com/neo-project/neo-modules/pull/902

## Type of change

- [x] Optimization

# How Has This Been Tested?

- [x] Current tests

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Jim8y — APPROVED** [review]

**Review by vncoelho — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Review by vncoelho — DISMISSED** [review]
Wait until coverall is merged otherwise tests are broken

**Review by vncoelho — APPROVED** [review]

**Pull request merged**
